src/tracks/playabletrack/wavetrack/ui/WaveTrackAffordanceControls.h

```diff
@@ -15,12 +15,14 @@
 
 class AffordanceHandle;
 class WaveClip;
+class TrackPanelResizeHandle;
+class WaveTrackAffordanceHandle;
 
 class AUDACITY_DLL_API WaveTrackAffordanceControls : public CommonTrackCell
 {
     std::weak_ptr<WaveClip> mFocusClip;
-    std::weak_ptr<AffordanceHandle> mAffordanceHandle;
-    std::weak_ptr<UIHandle> mResizeHandle;
+    std::weak_ptr<WaveTrackAffordanceHandle> mAffordanceHandle;
+    std::weak_ptr<TrackPanelResizeHandle> mResizeHandle;
```

Comment:
I see the purpose of this change and the commit that introduces it is to fix a "slicing problem" in AssignUIHandlePtr, where an object of derived type may be assigned to an object of base type, losing information.  Yes?

I don't know why I didn't see this earlier in review!

I suggest this additional change (add it to the label refactoring branch maybe):

In AssignUIHandlePtr, in the else-branch, add

```
      wxASSERT(typeid(*ptr) == typeid(*pNew));
```
You may also need
```
#include <typeinfo>
```

Reply:
This suggestion wasn't implemented yet